data/connection.go

```diff
@@ -55,7 +56,7 @@ func (c *PostgresSQL) GetProducts() (model.Coffees, error) {
 
 	// fetch the ingredients for each coffee
 	for n, cof := range cos {
-		i := []model.CoffeeIngredients{}
+		i := []model.CoffeeIngredient{}
 		err := c.db.Select(&i, "SELECT ingredient_id FROM coffee_ingredients WHERE coffee_id=$1", cof.ID)
```

Comment:
do we want to filter on ingredients where quantity > 0?

```suggestion
		err := c.db.Select(&i, "SELECT ingredient_id FROM coffee_ingredients WHERE coffee_id=$1 AND quantity > 0", cof.ID)
```

Reply:
Oh yeah! good one.